integration/testrunner/common.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Lumberjack authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testrunner
+
+import (
+	"net/mail"
+	"testing"
+
+	"cloud.google.com/go/logging/apiv2/loggingpb"
+)
+
+func diffLogEntry(tb testing.TB, logEntry *loggingpb.LogEntry, requireJustification bool) {
+	tb.Helper()
+
+	tb.Logf("Got logEntry is %s", logEntry.String())
+
+	jsonPayloadInfo := parseJSONPayload(tb, logEntry)
+
+	if logEntry.LogName == "" {
+		tb.Errorf("queryResult field %v is blank", "logName")
+	}
+	if jsonPayloadInfo.AuthenticationInfo.PrincipalEmail == "" || !validEmail(jsonPayloadInfo.AuthenticationInfo.PrincipalEmail) {
```

Comment:
Can we wrap `jsonPayloadInfo.AuthenticationInfo.PrincipalEmail == ""` in `validEmail` func? Also usually we do `isValidEmail` for funcs return boolean